topic-operator/src/main/java/io/strimzi/operator/topic/TopicSerialization.java

```diff
@@ -127,18 +129,38 @@ private static int getPartitions(KafkaTopic kafkaTopic) {
      */
     public static KafkaTopic toTopicResource(Topic topic, LabelPredicate resourcePredicate) {
         ResourceName resourceName = topic.getOrAsMapName();
-        return new KafkaTopicBuilder().withApiVersion("v1")
-                    .withMetadata(new ObjectMetaBuilder()
+        ObjectMeta om = topic.getMetadata();
+        if (om != null) {
+            om.setName(resourceName.toString());
+            Map<String, String> lbls = new HashMap<>();
+            lbls.putAll(resourcePredicate.labels());
+            if (topic.getMetadata().getLabels() != null)
+                lbls.putAll(topic.getMetadata().getLabels());
+            om.setLabels(lbls);
+            om.setOwnerReferences(topic.getMetadata().getOwnerReferences());
+            om.setAnnotations(topic.getMetadata().getAnnotations());
+        } else {
+            om = new ObjectMetaBuilder()
                     .withName(resourceName.toString())
-                    .withLabels(resourcePredicate.labels()).build())
-                    // TODO .withUid()
+                    .withLabels(resourcePredicate.labels())
+                    .build();
+        }
+
+        KafkaTopic kt = new KafkaTopicBuilder().withApiVersion("v1")
+                .withMetadata(om)
+                // TODO .withUid()
```

Comment:
What about this TODO? IS there an issue? Is that something what can be removed?

Reply:
That is not my change. I probably just changed indentation.

https://github.com/strimzi/strimzi-kafka-operator/blob/master/topic-operator/src/main/java/io/strimzi/operator/topic/TopicSerialization.java#L134